src/pkg/bundler/common.go

```diff
@@ -0,0 +1,309 @@
+func (b *Bundler) ExtractPackage(name string, out string) error {
+	message.Infof("Extracting %s to %s", name, out)
+	return nil
+	// read the index.json from the bfs.SourceTarball
+	// get all the layers for the package
+	// extract the layers to the output directory
+}
+
+// ValidateBundle validates the bundle
+func (b *Bundler) ValidateBundle() error {
+	if b.bundle.Metadata.Architecture == "" {
+		// ValidateBundle was erroneously called before CalculateBuildInfo
+		if err := b.CalculateBuildInfo(); err != nil {
+			return err
+		}
+		if b.bundle.Metadata.Architecture == "" {
+			return errors.New("unable to determine architecture")
+		}
+	}
+	if b.bundle.Metadata.Version == "" {
+		return fmt.Errorf("zarf-bundle.yaml is missing required field: metadata.version")
+	}
+	if b.bundle.Metadata.Name == "" {
+		return fmt.Errorf("zarf-bundle.yaml is missing required field: metadata.name")
+	}
+	if len(b.bundle.Packages) == 0 {
+		return fmt.Errorf("zarf-bundle.yaml is missing required list: packages")
+	}
+	for idx, pkg := range b.bundle.Packages {
+		message.Debug("Validating package:", message.JSONValue(pkg))
+
+		tmp, err := utils.MakeTempDir("")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(tmp)
+
+		if pkg.Repository == "" {
+			return fmt.Errorf("zarf-bundle.yaml .packages[%d] is missing required field: repository", idx)
+		}
+		if pkg.Ref == "" {
+			return fmt.Errorf("zarf-bundle.yaml .packages[%s] is missing required field: ref", pkg.Repository)
+		}
+		url := fmt.Sprintf("%s:%s-%s", pkg.Repository, pkg.Ref, b.bundle.Metadata.Architecture)
+		// validate access to packages as well as components referenced in the package
+		remote, err := oci.NewOrasRemote(url)
+		if err != nil {
+			// remote performs access verification upon instantiation
+			return err
+		}
+		if err := remote.PullPackageMetadata(tmp); err != nil {
+			return err
+		}
+		if err := packager.ValidatePackageSignature(tmp, pkg.PublicKey); err != nil {
+			return err
+		}
+		if len(pkg.OptionalComponents) > 0 {
+			// make sure if a wildcard is given, it is the first and only element
+			for idx, component := range pkg.OptionalComponents {
+				if (component == "*" && idx != 0) || (component == "*" && len(pkg.OptionalComponents) > 1) {
+					return fmt.Errorf("zarf-bundle.yaml .packages[%s].optional-components[%d] wildcard '*' must be first and only item", pkg.Repository, idx)
+				}
+			}
+			zarfYAML := types.ZarfPackage{}
+			zarfYAMLPath := filepath.Join(tmp, config.ZarfYAML)
+			err := utils.ReadYaml(zarfYAMLPath, &zarfYAML)
+			if err != nil {
+				return err
+			}
+			if pkg.OptionalComponents[0] == "*" {
+				// a wildcard has been given, so all optional components will be included
+				for _, c := range zarfYAML.Components {
+					if c.Only.Cluster.Architecture == "" || c.Only.Cluster.Architecture == b.bundle.Metadata.Architecture {
+						pkg.OptionalComponents = append(pkg.OptionalComponents, c.Name)
+					}
+				}
+				continue
+			}
```

Comment:
must also expand `aws-*` wildcard syntax here. Will fail if NO components satisfy arch. Continue if at least 1 satisfies and warn about the others?